src/EditorFeatures/CSharpTest2/Recommendations/RecommenderTests.cs

```diff
@@ -189,16 +193,52 @@ protected async Task VerifyAbsenceAsync(SourceCodeKind kind, string text)
             }
         }
 
-        protected string AddInsideMethod(string text)
+        protected string AddInsideMethod(string text, bool isAsync = false, string returnType = "void", bool topLevelStatement = false)
```

Comment:
not a huge lover of this.  i think it's fine to have a test say `AddInsideMethod("...")`.  where it gets really problematic for me is `AddInsideMethod("...", isAsync: true, returnType: ..., topLevelstatement: ...)`.  Ad this point i've lost the plot because my brain needs to stitch all that up to figure out what scenario is being tested.  just having the string *as is* is wonderfully clear and easy to grok without any extra mental cycles.

Reply:
Most cases only specify `topLevelStatement`, which is a huge improvement over multiplication of the tests themselves (where it becomes impossible to see if all cases are covered). I would not be opposed to an alternative _provided_ it supported test cases for both normal methods and top-level statements without requiring the content of the method to be duplicated.